### ISSUE TITLE
Stack corruption on arm64 when using DES methods

### ISSUE THREAD

**Issue by fake666** [x2on/OpenSSL-for-iPhone#126]
with the default build of the library, if you use DES methods on arm64, stack corruption will occur.

the reason is the definition of DES_LONG in opensslconf.h - the config header is taken from the first platform being built - x86_64 by default. for this platform however, DES_LONG is set to 'unsigned int', whereas for all other platforms it is set to 'unsigned long'.

thus you will be passing ints to a method that thinks it has longs it can write to.

to reproduce, run this on arm64:

```
DES_cblock key;
key[0] = 0x54; key[1] = 0x22; key[2] = 0x54; key[3] = 0x6a;
key[4] = 0x43; key[5] = 0x89; key[6] = 0xc8; key[7] = 0x66;

DES_key_schedule ks;
memset(&ks, 0, sizeof(ks));

uint8_t buf[200];
memset(buf, 0, 200);

DES_set_key_unchecked(&key, &ks);

for (int n = 0; n < 200; n++) {
    if (buf[n] != 0) {
        printf("stack corrupted!\n");
    }
}
```

**Comment by x2on** [comment]
Is this still open?

**Comment by oneton** [comment]
Quite sure it is, since it depends on PR #134.

**Comment by x2on** [comment]
Fixed by #142
